video/out/gpu/video.c

```diff
@@ -3095,16 +3101,6 @@ void gl_video_render_frame(struct gl_video *p, struct vo_frame *frame,
         p->ra->fns->clear(p->ra, target.tex, color, &target_rc);
     }
 
-    // The playloop calls this last before waiting some time until it decides
-    // to call flip_page(). Tell OpenGL to start execution of the GPU commands
-    // while we sleep (this happens asynchronously).
-    if ((p->opts.early_flush == -1 && !frame->display_synced) ||
-        p->opts.early_flush == 1)
-    {
-        if (p->ra->fns->flush)
-            p->ra->fns->flush(p->ra);
-    }
-
```

Comment:
Removal of this removes the behavior from opengl-cb. Must be fixed.

Reply:
What would be the best way of accomplishing this? I think we could get away with making `opengl-cb` reuse `ra_gl_ctx` to get the opengl-based options. That would also technically allow `opengl-cb` to get an option like `swapchain-depth` if that would be beneficial. In essence, `opengl-cb` would be its own “context provider”, using an external GL context.

Reply:
Maybe, but the API user does presentation himself when using opengl-cb. There's not much that would make sense to share. Just that the bit that makes sense (the early-flush logic) is now out of the common code and thus missing from opengl-cb.

Reply:
Okay. If the user does presentation himself then it's technically a case of an external swapchain, in which case we would set `gl->external_swapchain = true` and then the ra_gl_ctx helpers would “do the right thing” (specifically: `opengl-early-flush`, `opengl-finish` would still work. Technically even `opengl-waitvsync` and `opengl-check-pattern`, although it's questionable whether `opengl-cb` users benefit from enabling them)